### PR TITLE
Make the Main Menu's dialog buttons translatable

### PR DESCRIPTION
**Pull request by zenseii**
This makes it possible to translate the main menu buttons. More need to be added still and there should be some discussion on structure.

Relates to https://github.com/ihhub/fheroes2/issues/6105

### CODE REVIEW

**Comment by zenseii** [comment]
Hi, @ihhub.

I've added some buttons just to have example code that can be discussed. My biggest concern is how to preserve the original buttons from current asset and restore them when changing back from another language. Something similar to `OriginalAlphabetPreserver` could work.

**Comment by ihhub** [comment]
> Hi, @ihhub.
> 
> I've added some buttons just to have example code that can be discussed. My biggest concern is how to preserve the original buttons from current asset and restore them when changing back from another language. Something similar to `OriginalAlphabetPreserver` could work.
> 
> Please have a look at this when it suits you.

Hi @zenseii , the main question is do we want to show the original buttons at all? All original buttons are a part of ICN assets so they are present within the game. Based on the code we aren't using them at all.

**Comment by ihhub** [comment]
Hi @zenseii , then for these buttons if a selected language is the same as the assets we can add the code to retrieve images from the original ICNs. Something like this:
case ICN::NEW_GAME_BUTTON:
    if ( language == Language::Polish && crc32 == polishcrc32 ) {
        return fheroes2::AGG::GetICN( ICN::OriginalICN, id );
    }
...

**Comment by ihhub** [comment]
Hi @zenseii , to modify the size of space character please refer to getSpaceCharWidth() function in ui_text.cpp file.

File: src/fheroes2/gui/ui_language.cpp [hunk 1]
```diff
@@ -259,4 +259,12 @@ namespace fheroes2
 
         AGG::updateLanguageDependentResources( language, isOriginalResourceLanguage );
     }
+
+    SupportedLanguage getCurrentLanguage()
+    {
+        const Settings & conf = Settings::Get();
+
+        const fheroes2::SupportedLanguage currentLanguage = fheroes2::getLanguageFromAbbreviation( conf.getGameLanguage() );
```
Comment (ihhub) [inline src/fheroes2/gui/ui_language.cpp:267]: We can directly return the value of this function call without the need of a temporary variable.

File: src/fheroes2/agg/agg_image.cpp [hunk 2]
```diff
@@ -890,30 +1148,34 @@ namespace fheroes2
...
+                    if ( generateFrenchSpecificImages( id ) ) {
+                        return;
+                    }
+                    break;
+                case fheroes2::SupportedLanguage::Polish:
+                    if ( generatePolishSpecificImages( id ) ) {
+                        return;
+                    }
+                    break;
+                case fheroes2::SupportedLanguage::Italian:
```
Comment (ihhub) [inline src/fheroes2/agg/agg_image.cpp:1171]: What's about Russian? Is @Branikolog not happy to use the original resources?

**Review by ihhub — COMMENTED** [review]
Hi @zenseii , I left few comments here. Can you please check them?

File: src/fheroes2/agg/agg_image.cpp (hunk 2 above)
Comment (ihhub) [reply src/fheroes2/agg/agg_image.cpp:1171]: Ideally, we should remove all crappy buttons generated from other original buttons but sometimes they are too different from what we generate.

**Comment by ihhub** [comment]
@zenseii , huge thanks for these changes!

**Pull request merged**